### PR TITLE
167915672 Ft create an actual menu template

### PR DESCRIPTION
**Pull request by zellymiriam**
#### What does this PR do?
- It allows a user to create a menu template

#### Description of Task to be completed?
- add menuTemplateAction
- add menuTemplateReducer
- modify rootReducer
- modify MenuTemplate components
- add menuTemplateAction tests
- add menuTemplateReducer tests
- modify MenuTemplate component tests

#### How should this be manually tested?
- `git fetch` and `git checkout` to this branch
- Log in to the application
- Navigate to admin, click on  `Manage Menus` and then `Menu Templates`.
- Click on the `Create` button. The add menu template modal will pop up.
- Fill in the form to add a menu template

#### What are the relevant pivotal tracker stories?
- [#167915672](https://www.pivotaltracker.com/story/show/167915672)

#### Screenshots 
add_menu_template

### CODE REVIEW

**Comment by zellymiriam** [comment]
@AmosWels thanks for the feedback. I have worked on it you can review again.

File: client/src/components/Admin/Menus/MenuTemplate/index.js [hunk 1]
```diff
@@ -20,6 +25,13 @@ export class MenuTemplate extends Component {
     errors: {}
   }
 
+  componentWillReceiveProps(nextProps) {
```
Comment (azupatrick0) [inline client/src/components/Admin/Menus/MenuTemplate/index.js]: Hello @zellymiriam, please make use of getDerivedStateFromProps here as componentWillReceiveProps is no longer encouraged. Here => https://alligator.io/react/get-derived-state/

File: client/src/tests/components/Admin/Menus/MenuTemplate/index.test.js [hunk 2]
```diff
@@ -55,7 +62,22 @@ describe('Admin: Menu Component', () => {
...
     wrapper.instance().forceUpdate();
     wrapper.find('#select').simulate('change', event);
-    expect(spy).toBeCalled();
+  })
+
+  it('should submit data sucessfully  handle submit ', () => {
+    wrapper.find('button[name="add-btn"]').simulate('click');
+    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
 
+    wrapper.instance().forceUpdate();
```
Comment (azupatrick0) [inline client/src/tests/components/Admin/Menus/MenuTemplate/index.test.js]: Hello @zellymiriam, why are we forcing update here, is line 72 not meant to do the same?

File: client/src/components/Admin/Menus/MenuTemplate/index.js [hunk 3]
```diff
@@ -110,9 +132,23 @@ export class MenuTemplate extends Component {
             />
             )}
           </div>
+          <ToastContainer />
         </React.Fragment>
       );
     }
 }
+const mapStateToProps = (state) => ({
```
Comment (azupatrick0) [inline client/src/components/Admin/Menus/MenuTemplate/index.js]: Hello @zellymiriam, can we destructure the state here?

**Pull request merged**